PublishSubscribeDP/WeatherEventPublisher.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Weather_Monitoring.DataFormat;
+namespace Weather_Monitoring.PublishSubscribeDP
+{
+    public class WeatherEventPublisher
+    {
+        public void PublishEvent(string eventName, WeatherData weatherData)
+        {
+            if (WeatherEventSubscriber.subscribers.TryGetValue(eventName, out var subscribers))
+            {
+                foreach (var bot in WeatherEventSubscriber.subscribers[eventName])
```

Comment:
```suggestion
                foreach (var bot in subscribers)
```
that's why I suggest TryGetValue